src/screens/Departures/state/stop-place-details-state.ts

```diff
@@ -0,0 +1,135 @@
+} from 'use-reducer-with-side-effects';
+import {ErrorType, getAxiosErrorType} from '@atb/api/utils';
+import {getStopsDetails} from '@atb/api/departures/stops-nearest';
+import {StopsDetailsQuery} from '@atb/api/types/generated/StopsDetailsQuery';
+
+export type StopsDetailsDataState = {
+  data: StopsDetailsQuery | null;
+  error?: {type: ErrorType};
+  ids?: Array<string>;
+  isLoading: boolean;
+};
+
+const initialState: StopsDetailsDataState = {
+  data: null,
+  error: undefined,
+  ids: undefined,
+  isLoading: false,
+};
+
+type StopsDetailsDataActions =
+  | {
+      type: 'LOAD_DETAILS';
+      locations?: Array<string>;
+    }
+  | {
+      type: 'STOP_LOADER';
+    }
+  | {
+      type: 'UPDATE_STOP_DETAILS';
+      ids?: Array<string>;
+      result: StopsDetailsQuery;
+    }
+  | {
+      type: 'SET_ERROR';
+      error: ErrorType;
+    };
+
+const reducer: ReducerWithSideEffects<
+  StopsDetailsDataState,
+  StopsDetailsDataActions
+> = (state, action) => {
+  switch (action.type) {
+    case 'LOAD_DETAILS': {
+      if (!action.locations) return NoUpdate();
+      return UpdateWithSideEffect<
+        StopsDetailsDataState,
+        StopsDetailsDataActions
+      >(
+        {
+          ...state,
+          isLoading: true,
+          error: undefined,
+        },
+        async (state, dispatch) => {
+          if (!action.locations) return;
+          try {
+            const result = await getStopsDetails({ids: action.locations});
+            dispatch({
+              type: 'UPDATE_STOP_DETAILS',
+              ids: action.locations,
+              result,
+            });
+          } catch (e) {
+            dispatch({
+              type: 'SET_ERROR',
+              error: getAxiosErrorType(e),
+            });
+          } finally {
+            dispatch({type: 'STOP_LOADER'});
+          }
+        },
+      );
+    }
+
+    case 'STOP_LOADER': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        isLoading: false,
+      });
+    }
+
+    case 'UPDATE_STOP_DETAILS': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        isLoading: false,
+        ids: action.ids,
+        data: action.result,
+      });
+    }
+
+    case 'SET_ERROR': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        error: {
+          type: action.error,
+        },
+        data: state.data,
+      });
+    }
+
+    default:
+      return NoUpdate();
+  }
+};
+
+/***
+ * Use state for fetching details for a list of stopPlaces
+ *
+ * @param {Array<string>} [locationIds] - NSR ids to fetch details for
+ */
+export function useStopsDetailsData(locationIds?: Array<string>) {
+  const [state, dispatch] = useReducerWithSideEffects(reducer, initialState);
+
+  const refresh = useCallback(
+    () =>
+      dispatch({
+        type: 'LOAD_DETAILS',
+        locations: locationIds,
+      }),
+    [locationIds],
+  );
+
+  useEffect(refresh, [JSON.stringify(locationIds)]);
```

Comment:
Hmm. Hvorfor må `locationIds` gjøres om til string her, men ikke i dependency list til `useEffect` like over 🤔

Reply:
Ja, dette ser litt rart ut. Tror ikke du trenger refresh som callback når den er slik den er i det heletatt heller. Det kan noen ganger være fornuftig å caste om objekter til strenger for å få value comparison fremfor reference.

Reply:
Ojda, dette henger igjen fra noe feilsøking jeg drev med. Fikk testet på nytt uten stringify, og det ser ut til å fungerer bra.

Reply:
Angående `refresh` opprettet med `useCallback`: Er det ikke en risiko at noen brukere av denne hooken bruker `refresh` i dependency list til en hook, og at det fører til at den hooken blir kjørt ved hver render? 🤔 Generelt har jeg alltid brukt `useCallback` når jeg returnerer funksjoner fra hooks nettopp på grunn av dette, men det er kanskje jeg som ser spøkelser 😅